### DISCUSSION TITLE
你好,  我使用了 MQTT 推送了几个语音给兔子, 但是兔子不会按顺序播放, 好像会中断当前的 , 立刻播放收到的. 有没有办法然他按顺序播放呢?

### DISCUSSION THREAD

**Discussion by Tongvivi** [FoloToy/folotoy-server-self-hosting#84] [General | 讨论]
MQTT 功能方面的探讨
1. 推送语音,能否按顺序播放.
2. 有没有可能开放一些接口, 控制玩具的播放暂停, 音量大小 , 播放状态

关于语音对话方面
1. 兔子回答问题 总是断断续续的 ,每段停顿时间比较长. 我知道应该是服务器的延迟问题, 分成几段内容返回.  那么玩具能否做到, 在读第一段的同时, 准备好第二段内容.
谢谢!!

**Comment by lewangdev** [comment]
> MQTT 功能方面的探讨
>
> 推送语音,能否按顺序播放.

目前会立即播放当前发送的 url。我们考虑加一下播放列表。

> 有没有可能开放一些接口, 控制玩具的播放暂停, 音量大小 , 播放状态

这些功能我觉得也挺有用，我们看看怎么来安排开发

> 关于语音对话方面
> 
> 兔子回答问题 总是断断续续的 ,每段停顿时间比较长. 我知道应该是服务器的延迟问题, 分成几段内容返回. 那么玩具能否做到, 在读第一段的同时, 准备好第二段内容.

分句是越往后越长，在播放第一句的时候后面都准备好了，但是生成的内容是通过音频流拉取来进行播放，并不会下载到玩具的存储上，出现停顿时间比较长主要原因是下载过程比较慢，可以通过添加CDN等方式来改善
